### PR TITLE
xbps-src: hook to remove & warn cross cruft

### PR DESCRIPTION
**Pull request by sgn**
See #22139 

(I'll add as much as possible packages, but this PR can be merged as is)

I'll merge this if nobody complains by 2020-05-30

### CODE REVIEW

File: common/hooks/post-install/13-pkg-config-clean-xbps-cross-base-ref.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	fi
+	for f in "$PKGDESTDIR"/usr/lib/pkgconfig/*.pc \
+		"$PKGDESTDIR"/usr/share/pkgconfig/*.pc
+	do
+		if [ -f "$f" ]; then
+			# Sample sed script
+			# s,/usr/armv7l-linux-musleabihf/,/,g
+			# trailing / to avoid clashing with other $XBPS_CROSS_BASE and
+			# $XBPS_CROSS_TRIPLET reference.
+			sed -i -e "s,$XBPS_CROSS_BASE/,/,g" "$f"
```
Comment (sgn) [inline common/hooks/post-install/13-pkg-config-clean-xbps-cross-base-ref.sh:19]: tempted to change to:

    sed -i -e "s,$XBPS_CROSS_BASE/usr,/usr,g" "$f"

File: common/hooks/post-install/99-pkglint-warn-cross-cruft.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+hook() {
+	if [ -z "$CROSS_BUILD" ]; then
+		return 0
+	fi
+	for d in bin lib libexec share; do
+		for f in $PKGDESTDIR/usr/$d/* $PKGDESTDIR/usr/$d/**/*; do
+			case "$(file -bi "$f")" in
+			text/*) if grep -q -e "$XBPS_CROSS_BASE" \
+					   -e "$XBPS_WRAPPERDIR" "$f"; then
```
Comment (newbluemoon) [inline common/hooks/post-install/99-pkglint-warn-cross-cruft.sh:13]: Maybe also `$XBPS_CROSS_TRIPLET`?

Comment (sgn) [reply common/hooks/post-install/99-pkglint-warn-cross-cruft.sh:13]: I don't know.

```sh
git grep 'sed.*XBPS_CROSS_TRIP' srcpkgs/
```

returns no interesting use-cases.
Only used in language interpreter and compiler.

Comment (newbluemoon) [reply common/hooks/post-install/99-pkglint-warn-cross-cruft.sh:13]: One case I remember is wxWidgets.
Might be the only relevant case right now, but that could change with new packages.
Anyway, it’s just a hint printed out saying that there is something which should be checked.

Comment (sgn) [reply common/hooks/post-install/99-pkglint-warn-cross-cruft.sh:13]: Too much false positive will make people ignore warning.
If no other packages have problem with it, I don't think it's worth to warrant a warning.

I'd like to hear other's opinion. :)

Comment (newbluemoon) [reply common/hooks/post-install/99-pkglint-warn-cross-cruft.sh:13]: Fair point. :)

**Pull request merged**